MauroDataMapper/mdm-explorer
# gh-203 Create pages to view and create requests from templates

**Note:** this PR depends on MauroDataMapper-Plugins/mdm-plugin-explorer#14

Uses configuration from `mdm-plugin-explorer` to list any pre-made requests from a Mauro root folder and fork (copy) them as new user data requests.

- New top-level navigation "Templates" link, index and detail pages created
- Creation of shared, common components to make Template and My Request pages consistent
- New mdm-header-and-content-box component to simplify layout of dark green header (and action bar) with lighter content
- Make the label of a template/request row a clickable link to go to the detail page
- Refactor My Requests pages to re-use code and make constsient
- Make similar changes to the mdm-data-element-search-result component for consistency
- Add help content in pages to direct users to templates page
- Update documentation in /docs folder

Resolves #203 

# Testing

If you don't want to wait for MauroDataMapper-Plugins/mdm-plugin-explorer#14 to be merged, it is possible to test this PR locally:

1. Fetch and checkout the branch for MauroDataMapper-Plugins/mdm-plugin-explorer#14.
2. Build the plugin manually using `gradle publishToMavenLocal` to build your own local plugin snapshot
3. Use `mdm-application-build` and include a reference to `mdm-plugin-explorer` snapshot, this should pick up your local build copy
4. When running the backend, the plugin should now have bootstrapped a new folder called "Mauro Explorer Templates"
5. In MDE, create some data requests and submit them to finalise.
6. Move these submitted requests to the "Mauro Explorer Templates" folder (using MDM UI).
7. Finally, go to the `/templates` page in MDE to view the list of templates available.

# Screenshots

![image](https://user-images.githubusercontent.com/3219480/213732732-214fdf5c-d449-4e11-bbc5-185ea19f3e6f.png)

![image](https://user-images.githubusercontent.com/3219480/213732815-ff4cd031-3c23-4893-9dc9-81d5cb97aa57.png)
